### ISSUE TITLE
Error on recurring donations "Received unknown parameter: name"

### ISSUE THREAD

**Issue by danielstrum** [drastik/com.drastikbydesign.stripe#248]
Hi... 
This is a brand new CiviCRM installation on a Wordpress 4.9.4 site. Everything except Recurring donations works fine. On recurring donations, I get the error:
**Received unknown parameter: name**

Using API version 2018-02-06 and extension version 4.7.3.

Thanks for the great extension... I use it on 3 other sites I manage.

**Comment by danielstrum** [comment]
Hmmm.... Looking at the API changelog I see from the 2018-02-05 upgrade: 

> Major Each plan object is now linked to a product object with type=service. The plan object fields statement_descriptor and **name** attributes have been moved to product objects. Creating a plan now requires passing a product attribute to POST /v1/plans. This may be either an existing product ID or a dictionary of product fields, so that you may continue to create plans without separately creating products.

**Comment by petednz** [comment]
Just trying this out and hit same error. Can hopefully get some digging done later today

**Comment by danielstrum** [comment]
Thank You!

**Comment by petednz** [comment]
Logs showed: /sites/default/civicrm/extensions/com.drastikbydesign.stripe/CRM/Stripe/Page/Webhook.php(83): CRM_Core_Error::fatal("Stripe Callback: cannot json_decode data, exiting. 

On my site i was testing recurring and had not set the Webhook Endpoint per the instructions. Having done that I no longer get the error and I see a payment in stripe. 

I don't see anything in stripe that suggests that payment is a recurring payment - but i might let that sit and see what happens tomorrow.

@danielstrum did you have your webhook set?

**Comment by petednz** [comment]
Hmm just put another through and got same error. Noting that the one that didn't fail shows in civicrm as not recurring so maybe i mis set the page

**Comment by danielstrum** [comment]
Yes... I set my Webhook Endpoint. Did you see the changelog for the API that I noted above (my first comment).

> Major Each plan object is now linked to a product object with type=service. The plan object fields statement_descriptor and name attributes have been moved to product objects. Creating a plan now requires passing a product attribute to POST /v1/plans. This may be either an existing product ID or a dictionary of product fields, so that you may continue to create plans without separately creating products.

**Comment by petednz** [comment]
Yes - will get @jitendrapurohit to take a look when he clocks in

**Comment by danielstrum** [comment]
Yes, this seems to fix the issue. Thanks!

**Comment by mpwoodward** [comment]
I'm new to CiviCRM and this extension both -- I'm on version 4.7.3 and getting this error. Just wanted to be clear on whether or not it should be fixed in 4.7.3, or if I need to install 4.7-dev to get around the error? Thanks.

**Comment by mpwoodward** [comment]
Never mind -- installed 4.7-dev and it's working now. Some wrinkles with the confirmation page and email receipt in live mode (works in test) but that probably isn't related to the Stripe extension. Thanks for your work on this!

**Comment by highfalutin** [comment]
Hi all, I'm encountering the same error on version 4.7.3 of the extension. Are you all using 4.7-dev on your production sites to get past this error? Any idea when we can expect a new official release?

**Comment by mwoodward-atd** [comment]
I'm using 4.7-dev in production and it's working fine.
